sdr/liquid_dsp/Makefile

```diff
@@ -0,0 +1,247 @@
+###########################################################################
+# apps/sdr/liquid_dsp/Makefile
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################/
+
+include $(APPDIR)/Make.defs
+
+LIQUID_DSP_VERSION  = 1.4.0
+LIQUID_DSP_UNPACK   = liquid-dsp
+LIQUID_DSP_TARBALL  = v$(LIQUID_DSP_VERSION).tar.gz
+LIQUID_DSP_URL_BASE = https://github.com/jgaeddert/liquid-dsp/archive
+LIQUID_DSP_URL      = $(LIQUID_DSP_URL_BASE)/$(LIQUID_DSP_TARBALL)
+
+$(LIQUID_DSP_TARBALL):
+	$(Q) echo "Downloading $(LIQUID_DSP_TARBALL)"
+	$(Q) curl -O -L $(LIQUID_DSP_URL)
+
+$(LIQUID_DSP_UNPACK): $(LIQUID_DSP_TARBALL)
+	$(Q) echo "Unpacking $(LIQUID_DSP_TARBALL) to $(LIQUID_DSP_UNPACK)"
+	$(Q) tar xzvf $(LIQUID_DSP_TARBALL)
+	$(Q) mv liquid-dsp-$(LIQUID_DSP_VERSION) $(LIQUID_DSP_UNPACK)
+	$(Q) cp config.h $(LIQUID_DSP_UNPACK)/include
+
+$(LIQUID_DSP_UNPACK)/.patch: $(LIQUID_DSP_UNPACK)
+	$(Q) touch $(LIQUID_DSP_UNPACK)/.patch
+
+VPATH += $(LIBUV_UNPACK)/include
+
+DEPPATH += --dep-path $(LIBUV_UNPACK)/include
+
+CFLAGS += -I$(LIQUID_DSP_UNPACK)/include
+CFLAGS += -Wall -std=c99
+CFLAGS += -DM_SQRT1_2=0.7071067811865475244008443621048490
+CFLAGS += -DM_SQRT2=1.4142135623730950488016887242096981
+CFLAGS += -DM_PI=3.1415926535897932384626433832795029
+CFLAGS += -DM_LN2=0.6931471805599453094172321214581765
+CFLAGS += -DM_PI_2=1.5707963267948966192313216916397514
+CFLAGS += -DM_2_PI=0.6366197723675813430755350534900574
```

Comment:
let's move the common macro to:
https://github.com/apache/nuttx/blob/master/include/nuttx/lib/math.h#L106-L126

Reply:
> let's move the common macro to: https://github.com/apache/nuttx/blob/master/include/nuttx/lib/math.h#L106-L126

Those lines in math.h are included only if libm is enabled. What if this is not the case? Libm from NuttX has quite some problems. When working on LuaJIT support, I have found out that libm stuff doesn't work as it should. Because pressed with time I have sticked with builtin compiler math lib.

Reply:
> > let's move the common macro to: https://github.com/apache/nuttx/blob/master/include/nuttx/lib/math.h#L106-L126
> 
> Those lines in math.h are included only if libm is enabled. What if this is not the case? 

so these macro doesn't exist in toolchain's libm?

> Libm from NuttX has quite some problems. When working on LuaJIT support,

Do you have a workable LuaJIT? it's very great if you can upstream the porting.

> I have found out that libm stuff doesn't work as it should. Because pressed with time I have sticked with builtin compiler math lib.

it should be fixed with the recent change(report from @fjpanag):
https://github.com/apache/nuttx/pull/8495

Reply:
> Do you have a workable LuaJIT? it's very great if you can upstream the porting.

Yes, I have a working version, but not without some changes in NuttX. I still need to fix that first. One of problems is a new UNUSED macro in NuttX, which is giving me problems with LuaJIT code. The old version of UNUSED macro was fine, but then someone changed it. I am planning to release LuaJIT support when EdgePro1 campaign will start.

Reply:
When does the campaign start?

Reply:
I hope it will start soon. It si quite some administrative work to do before launch. Will see if I can release LuaJIT prior to campign launch?

Reply:
> so these macro doesn't exist in toolchain's libm?

They should be, but if I remove the definitions from my Makefile, i am getting those undefined errors. Maybe you can take a look at it?

Reply:
> Sorry, it's changed by me, let's see how to make it work with LuaJIT. Once we fix the problem, let's enable LuaJIT in CI to catch the potential break before merging the patch.

Let me open PR in the near future and maybe we can work together to bring LuaJIT to life. I am also relying on some archiutecture specific stuff in my port, i.e. ITCM memory heap for JIT traces, because I am running LuaJIT on i.MX RT1064.